src/v/datalake/protobuf_to_arrow_converter.h

```diff
@@ -22,12 +22,18 @@
 
 #include <memory>
 #include <stdexcept>
+#include <string_view>
```

Comment:
nit: i wonder if this should go in the #if block too, though i guess it won't matter until we have a strict IWYU policy

Reply:
The old code will be deleted before then, and this will be required :)